plugins/scaffolder/src/components/fields/RepoUrlPicker/utils.ts

```diff
@@ -44,22 +44,22 @@ export function serializeRepoPickerUrl(data: RepoUrlPickerState) {
 export function parseRepoPickerUrl(
   url: string | undefined,
 ): RepoUrlPickerState {
-  let host = undefined;
-  let owner = undefined;
-  let repoName = undefined;
-  let organization = undefined;
-  let workspace = undefined;
-  let project = undefined;
+  let host = '';
```

Comment:
Had to change all these values from `undefined` to `''` as these values are passed to `mui` inputs and they are controlled components. An error is thrown if the component has a controlled value that moves from `undefined` to `''` so should follow something similar to this: https://reactjs.org/docs/forms.html#controlled-components

Reply:
```
Warning: ComponentName is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component
```